packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -34,6 +34,56 @@ typedef ScrollableWidgetBuilder = Widget Function(
   ScrollController scrollController,
 );
 
+class DraggableScrollableSheetController {
+  DraggableScrollableSheetController();
+
+  late _DraggableScrollableSheetScrollController _scrollController;
+
+  void _attach(_DraggableScrollableSheetScrollController scrollController) {
+    _scrollController = scrollController;
+  }
+
+  void reset() {
+    _scrollController.extent.currentExtent =
+        _scrollController.extent.initialExtent;
+  }
+
+  List<double> get snapSizes => _scrollController.extent.snapSizes;
+
+  double pixelsToExtent(double pixels) =>
+      _scrollController.extent.pixelsToExtent(pixels);
+
+  void animateTo(
+    double offset, {
+    Duration duration = const Duration(milliseconds: 200),
+    Curve curve = Curves.easeOut,
+  }) {
+    final AnimationController controller = AnimationController.unbounded(
+      vsync: _scrollController.position.context.vsync,
+      value: _scrollController.extent.currentExtent,
+    );
+    CurvedAnimation(parent: controller, curve: curve).addListener(() {
+      _scrollController.extent.updateExtent(controller.value,
+          _scrollController.position.context.notificationContext!);
+    });
+    controller
+        .animateTo(offset, duration: duration)
```

Comment:
As is, this animation will not be interruptible by user input-this is probably undesirable and differs from `ScrollController.animateTo`, which is interuptible.

Probably best is to use `position.beginActivity` with a `DrivenActivity` here. This code will be simpler and gives us interruptibility for free.

See [ScrollPositionWithSingleContext.animateTo](https://github.com/flutter/flutter/blob/ff15d04f214ae7cc4ffbacde2faefa1772e112e4/packages/flutter/lib/src/widgets/scroll_position_with_single_context.dart#L170) for how to do this.

We probably also want to copy their logic for skipping the animation if the sheet is already within tolerance of the target.

We could just directly call `animateTo` on the sheet's generic `ScrollController` that it passes in it's builder, but that'll fail if the developer supplied builder hasn't attached a scrollable to the controller yet, so that's probably a non starter. Not sure if I remember right, but I think I included a test to ensure nothing throws if you interact with a sheet that has no child scrollable. If that test doesn't exist, we should probably add it in this PR or a separate one.